cpp/src/arrow/ipc/feather.cc

```diff
@@ -558,8 +577,18 @@ class TableWriter::TableWriterImpl : public ArrayVisitor {
       // byte boundary, and we write this much data into the stream
       int64_t null_bitmap_size = GetOutputLength(BitUtil::BytesForBits(values.length()));
       if (values.null_bitmap()) {
-        RETURN_NOT_OK(WritePadded(stream_.get(), values.null_bitmap()->data(),
-                                  null_bitmap_size, &bytes_written));
+        auto null_bitmap = values.null_bitmap();
+        if (values.offset() > 0) {
+          if (!pool_) {
+            return Status::Invalid(
+                "Require memory pol to write array slice (i.e. offset > 0) with nulls.");
+          }
+          RETURN_NOT_OK(GetTruncatedBitmap(values.offset(), values.length(), null_bitmap,
+                                           pool_, &null_bitmap));
+        }
+
+        RETURN_NOT_OK(WritePadded(stream_.get(), null_bitmap->data(), null_bitmap_size,
```

Comment:
Wouldn't it be better to extend `WritePadded` or write a second version of it that can handle bit-offsets? Using that way we could avoid the memory allocation for `GetTruncatedBitmap` above.

Reply:
I agree that would be better if it is possible. Same goes for the arrow writing code of course. I can give it a try tomorrow.

Reply:
I've pushed a change to do as you suggested and added a test.